compiler/p/codegen/OMRMemoryReference.hpp

```diff
@@ -76,6 +76,20 @@ class OMR_EXTENSIBLE MemoryReference : public OMR::MemoryReference
    uint8_t _flag;
    uint8_t _length;
 
+   protected:
+
+   // The extra int parameter at the end is needed to ensure that this constructor is never
+   // ambiguous with the legacy constructor taking an int32_t for its displacement. This detail
+   // is hidden from other code, since this constructor can only be used by calling the
+   // MemoryReference::withDisplacement helper. Once the legacy constructor is removed, the extra
+   // parameter here can also be removed.
+   MemoryReference(
+         TR::Register *br,
+         int64_t disp,
+         uint8_t len,
+         TR::CodeGenerator *cg,
+         int);
```

Comment:
Can we assume this cleanup will be performed as part of #4720 or some other issue?

Reply:
This cleanup would have to happen with/after #5172, which would switch away from direct use of all `MemoryReference` constructors in favour of static factory methods and thus eliminate the need for the old constructor to continue to exist.